src/condor_tests/test_htcondor_cli_sys_status.py

```diff
@@ -0,0 +1,183 @@
+    {
+        "MyAddress" : "FILLER",
+        "Machine" : "LOCAL_HOST",
+        "MyType" : "Scheduler",
+        "RecentDaemonCoreDutyCycle" : 0,
+        "TransferQueueNumUploading" : 1,
+        "FileTransferUploadBytesPerSecond_1m" : 0,
+        "ExpectedDaemon" : "SCHEDD",
+        "ExpectedHP" : 93,
+    },
+    # Verify Download MB Waiting < 100 -> 100 HP
+    {
+        "MyAddress" : "FILLER",
+        "Machine" : "LOCAL_HOST",
+        "MyType" : "Scheduler",
+        "RecentDaemonCoreDutyCycle" : 0,
+        "FileTransferMBWaitingToDownload" : 10,
+        "TransferQueueDownloadWaitTime" : 100000,
+        "TransferQueueNumWaitingToDownload" : 3,
+        "ExpectedDaemon" : "SCHEDD",
+        "ExpectedHP" : 100,
+    },
+    # Verify Download MB Waiting > 100 && Wait time < 5m -> 100 HP
+    {
+        "MyAddress" : "FILLER",
+        "Machine" : "LOCAL_HOST",
+        "MyType" : "Scheduler",
+        "RecentDaemonCoreDutyCycle" : 0,
+        "FileTransferMBWaitingToDownload" : 250,
+        "TransferQueueDownloadWaitTime" : 20,
+        "TransferQueueNumWaitingToDownload" : 3,
+        "ExpectedDaemon" : "SCHEDD",
+        "ExpectedHP" : 100,
+    },
+    # Verify Download MB Waiting > 100 && Wait time > 5m -> 93 HP
+    {
+        "MyAddress" : "FILLER",
+        "Machine" : "LOCAL_HOST",
+        "MyType" : "Scheduler",
+        "RecentDaemonCoreDutyCycle" : 0,
+        "FileTransferMBWaitingToDownload" : 250,
+        "TransferQueueDownloadWaitTime" : 100000,
+        "TransferQueueNumWaitingToDownload" : 3,
+        "ExpectedDaemon" : "SCHEDD",
+        "ExpectedHP" : 93,
+    },
+]
+
+@action(params={name: name for name in TEST_CASES})
+def run_commands(default_condor, request):
+    with default_condor.use_config():
+        # HTCondor cli logger outputs to stderr?
+        p = subprocess.run(TEST_CASES[request.param], stdout=subprocess.PIPE, stderr=subprocess.PIPE, timeout=60)
+        output = p.stdout.rstrip().decode() + p.stderr.rstrip().decode()
+        yield output
+
+class TestHTCondorCLISytemStatus:
+    def test_convert_ad_unit_tests(self):
+        for test_ad in UNIT_TEST_CASES:
+            display = "\nFailed Unit Test Input Ad: {{\n{0}\n}}\n".format('\n'.join([f'\t{key} : {val}' for key, val in test_ad.items()]))
+            result = _ad_to_daemon_status(test_ad)
+            if "MyType" not in test_ad or test_ad["MyType"] == "Shadow":
+                assert result is None, display
+            else:
+                daemon, status = result
+                assert test_ad["ExpectedDaemon"] == daemon, display
+                assert test_ad["ExpectedHP"] == int(status["HealthPoints"]), display
+
+    def test_htcondor_cli_sys_status(self, run_commands):
+        # Ignore the first line (header)
+        for line in run_commands.split("\n")[1:]:
+            if line.strip() != "":
+                # Common for Master Ad to not be reported -> UNKNOWN status
```

Comment:
This seems worrisome, but for unrelated reasons.

Reply:
I had same thought... will say in my local testing it seemed consistent that the Master daemon had an unknown status. Maybe I should add a 5-10 second sleep to ensure the mini-condor is fully up and reporting, but that seems like fixing a symptom so IDK

Reply:
Also, we are seeing intermittent failures because the Collector has a low `RecentDaemonCoreDutyCycle` (in the `Bad` range according to the code)

Reply:
One of the reasons I wanted to test a live, if idle, example was because while we're used to looking at pools and pool components which are expected to be busy all the time, I'm not at all sure that the kind of user who would benefit from this tool will be.  I don't presently have any good ideas about how to report this.

Reply:
Note: @Todd-L-Miller you did not misread this... I miss typed. I meant to say we must be getting a high `RecentDaemonCoreDutyCycle` because the returned health is `Bad` in the test suite

Reply:
I think this issue was a result of the first Collector Ad stored has a high `RecentDaemonCoreDutyCycle` at startup and is queried very quickly by the test. Setting `COLLECTOR_UPDATE_INTERVAL` to a small number (2) appears to resolve the issue.